### ISSUE TITLE
default-extensions in .cabal ignored

### ISSUE THREAD

**Issue by turion** [ndmitchell/hlint#1504]
I have a project with lots of modules needing the `Arrows` extension, so I enabled it globally by putting it in the `default-extensions` field in the `.cabal` file. But now `hlint` does not detect the presence of this extension anymore, and complains about a syntax error although there is none.

I tried globally enabling that extension for `hlint` by adding this to `.hlint.yaml`, with no effect:

```
- extensions:
    - name: [Arrows]
```

I finally managed to work around the issue by adding this to `.hlint.yaml`:

```
- arguments: [-XArrows]
```

Weirdly, this happens for `Arrows`, but not for other syntax-changing extension such as `TupleSections`.

**Comment by zliu41** [comment]
This is explained in the readme: https://github.com/ndmitchell/hlint#language-extensions

`Arrows` is one of those non-backwards compatible extensions, meaning enabling `Arrow` may cause a module to fail to compile. This is not the case with `TupleSections`.

**Comment by turion** [comment]
I understand the design choice, still I think it would be cleverer if `hlint` would try to detect the actually used extensions. Otherwise a benign refactoring like moving extensions from the source files to the cabal file break it.
